src/test/java/spoon/test/enums/EnumsTest.java

```diff
@@ -16,136 +16,139 @@
-
-		final CtEnum<?> foo = (CtEnum) launcher.getFactory().Type().get(Foo.class);
-		assertEquals(1, foo.getFields().size());
-		assertEquals(1, foo.getFields().get(0).getAnnotations().size());
+	void testAnnotationsOnEnum() throws Exception {
+		// contract: an annotation on an enum value is represented in the spoon model
+		final CtEnum<?> foo = build("spoon.test.enums.testclasses", "Foo");
+		assertThat(foo.getEnumValues().size(), is(1));
+		CtEnumValue<?> value = foo.getEnumValues().get(0);
+		assertThat(value.getAnnotations().size(), is(1));
+		CtAnnotationType<?> annotation = (CtAnnotationType<?>) foo.getFactory().Annotation().<Deprecated>get(Deprecated.class);
+		assertThat(map(AnnotationTest::getActualClassFromAnnotation, value.getAnnotations()), hasItem(Deprecated.class));
 		assertSame(Deprecated.class, AnnotationTest.getActualClassFromAnnotation(
 				foo.getFields().get(0).getAnnotations().get(0)));
-		assertEquals(
-				"public enum Foo {" + DefaultJavaPrettyPrinter.LINE_SEPARATOR + DefaultJavaPrettyPrinter.LINE_SEPARATOR
-						+ "    @java.lang.Deprecated"
-						+ DefaultJavaPrettyPrinter.LINE_SEPARATOR + "    Bar;}",
-				foo.toString());
+		// finding with a regex to avoid printer-specific output
+		assertThat(foo.prettyprint(), containsRegexMatch("@(java\\.lang\\.)?Deprecated\\W+Bar"));
+	}
+
+	private static <I, O> List<O> map(Function<I, O> function, List<I> input) {
+		return input.stream().map(function).collect(Collectors.toList());
 	}
 
 	@Test
-	public void testEnumWithoutField() throws Exception {
+	void testEnumWithoutValue() throws Exception {
+		// contract: an enum without values contains a ; before any other members
 		final Factory factory = build(Burritos.class);
 		final CtType<Burritos> burritos = factory.Type().get(Burritos.class);
-		assertEquals("public enum Burritos {" + DefaultJavaPrettyPrinter.LINE_SEPARATOR //
-				+ "    ;" + DefaultJavaPrettyPrinter.LINE_SEPARATOR + DefaultJavaPrettyPrinter.LINE_SEPARATOR //
-				+ "    public static void m() {" + DefaultJavaPrettyPrinter.LINE_SEPARATOR //
-				+ "    }" + DefaultJavaPrettyPrinter.LINE_SEPARATOR //
-				+ "}", burritos.toString());
+		assertThat(burritos.prettyprint(), containsRegexMatch("Burritos \\{\\W+;"));
 	}
 
 	@Test
-	public void testGetAllMethods() throws Exception {
+	void testGetAllMethods() throws Exception {
 		// contract: getAllMethods also returns the methods of Enum
 		final Factory factory = build(Burritos.class);
 		final CtType<Burritos> burritos = factory.Type().get(Burritos.class);
-		CtMethod name = factory.Core().createMethod();
+		CtMethod<String> name = factory.Core().createMethod();
 		name.setSimpleName("name"); // from Enum
 		name.setType(factory.Type().createReference(String.class));
 		assertTrue(burritos.hasMethod(name));
 		assertTrue(burritos.getAllMethods().contains(name));
+		// this does not work due to the SignatureBasedSortedSet violating the Set contract
+		// assertThat(burritos.getAllMethods(), hasItem(name));
```

Comment:
This comment is noise for this test, and it's very unlikely to be removed if `SignatureBasedSortedSet` is fixed. I'd remove it.